src/azure/cli/commands/arm.py

```diff
@@ -150,3 +151,201 @@ def required_values_validator(namespace):
+                    raise CLIError('--add should be of the form:'
+                                   ' --add property.list key1=value1 key2=value2')
+            elif arg_type == '--remove':
+                try:
+                    remove_properties(instance, expressions)
+                except ValueError:
+                    raise CLIError('--remove should be of the form: --remove'
+                                   ' property.propertyToRemove or'
+                                   ' --remove property.list <indexToRemove>')
+            else:
+                raise ValueError('Unsupported arg type {}'.format(arg_type))
+
+        # Done... update the instance!
+        getterargs[setter_arg_name] = instance
+        opres = setter(**getterargs)
+        return opres.result() if isinstance(opres, AzureOperationPoller) else opres
+
+    class OrderedArgsAction(argparse.Action): #pylint:disable=too-few-public-methods
+        def __call__(self, parser, namespace, values, option_string=None):
+            if not getattr(namespace, 'ordered_arguments', None):
+                setattr(namespace, 'ordered_arguments', [])
+            namespace.ordered_arguments.append((option_string, values))
+
+    def one_required(namespace):
+        if not getattr(namespace, 'ordered_arguments', None):
+            raise ValueError('At least one must be specified: --add, --set, --remove')
+
+    cmd = CliCommand(name, handler)
+    cmd.arguments.update(set_arguments)
+    cmd.arguments.update(get_arguments)
+    cmd.arguments.pop(setter_arg_name, None)
+    cmd.add_argument('properties_to_set', '--set', nargs='+', action=OrderedArgsAction, default=[],
+                     help='Update an object by specifying a property path and value to set.'
+                     '  Example: --set property1.property2=value',
+                     metavar='PROPERTY1.PROPERTY2=VALUE')
+    cmd.add_argument('properties_to_add', '--add', nargs='+', action=OrderedArgsAction, default=[],
+                     help='Add an object to a list of objects by specifying a path and key'
+                     ' value pairs.  Example: --add property.list key=<value>',
+                     metavar='PROPERTY.LIST KEY=<VALUE>')
+    cmd.add_argument('properties_to_remove', '--remove', nargs='+', action=OrderedArgsAction,
+                     default=[], help='Remove a property or an element from a list.  Example: '
+                     '--remove property.list <index>', metavar='PROPERTY.LIST <INDEX>',
+                     validator=one_required)
+    command_table[name] = cmd
+
+index_regex = re.compile(r'\[(.*)\]')
+def set_properties(instance, expression):
+    key, value = expression.split('=', 1)
+
+    #pylint:disable=redefined-variable-type
+    if value in ('{}', '{ }'):
+        value = {}
+    elif value in ('[]', '[ ]'):
+        value = []
+    elif value.lower() == 'false':
+        value = False
+    elif value.lower() == 'true':
+        value = True
```

Comment:
I switched to parsing a json doc, which handles these cases and gives us the first step in full JSON doc support (now supports a full JSON payload for value)

---
In reply to: [72830828](https://github.com/Azure/azure-cli/pull/558#discussion_r72830828) [](ancestors = 72830828)